### PR TITLE
SUSE Manager - Navigate from patch detail to previews pages

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
Right now the patch detail view brings back the user to the host details page upon clicking the back link.

This PR implements a lookup on the history stack to determine if we have enough history to go back to the previous page or navigate back to related patches page by default.

## How was this tested?
Jest tests added.

### CODE REVIEW

File: assets/js/lib/history/index.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// HANDLE WITH CARE
+export const historyLength = (windowImpl = window) => windowImpl.history.length;
```
[the remark this answers is not in the document]
Comment (balanza) [reply assets/js/lib/history/index.js:2]: image

```suggestion
export const historyLength = ({ history } = window) => history.length;
```

[the remark this answers is not in the document]
Comment (balanza) [reply assets/js/lib/history/index.js:2]: tbh, the name I don't like is `historyLength`. I suggest something like

```js
// history/index.js
export const length = ({ history } = window) => history.length;

// page.jsx
import * as history from '@/lib/history'
history.length()
```

File: assets/js/pages/AdvisoryDetails/AdvisoryDetailsPage.jsx [hunk 2]
```diff
@@ -22,10 +24,20 @@ function AdvisoryDetailsPage() {
 
   return (
     <>
-      <BackButton url={`/hosts/${hostID}`}>Back</BackButton>
-      {!isLoading ? (
+      <BackButton
+        onClick={() => {
```
Comment (EMaksy) [inline assets/js/pages/AdvisoryDetails/AdvisoryDetailsPage.jsx:28]: I always prefer the slightly clearer readability of ternary operator
```
<BackButton
  onClick={() => 
    historyLength() < 3 
      ? navigate(`/hosts/${hostID}/patches`) 
      : navigate(-1)
  }
 ```

File: assets/js/lib/history/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (EMaksy) [reply assets/js/lib/history/index.js:2]: I like to keep the module, one module one test, just helps not to forget :)

about the naming... well it is the hardest thing in this job.... so i like the meme haha.

**Review by EMaksy — APPROVED** [review]
Hey the code lgtm, added just some comments/ideas :+1:

**Pull request merged**